pybedlite/tests/test_overlap_detector.py

```diff
@@ -216,6 +219,144 @@ def test_construction_from_ucsc_with_strand(strand: str) -> None:
+            return False
+
+    @dataclass(eq=True, frozen=True)
+    class ZeroBasedUnstranded:
+        refname: str
+        zero_based_start: int
+        end: int
+
+        @property
+        def start(self) -> int:
+            """A 0-based start position."""
+            return self.zero_based_start
+
+    @dataclass(eq=True, frozen=True)
+    class ZeroBasedStranded:
+        refname: str
+        zero_based_start: int
+        end: int
+        negative: bool
+
+        @property
+        def start(self) -> int:
+            """A 0-based start position."""
+            return self.zero_based_start
+
+    # Create minimal features of all supported structural types
+    zero_based_protocol = ZeroBasedOpenEndedProtocol(refname="chr1", start=1, end=50)
+    one_based_protocol = OneBasedProtocol(contig="chr1", one_based_start=10, end=60)
+    zero_based_unstranded = ZeroBasedUnstranded(refname="chr1", zero_based_start=20, end=70)
+    zero_based_stranded = ZeroBasedStranded(
+        refname="chr1", zero_based_start=30, end=80, negative=True
+    )
+    # Set up an overlap detector to hold all the features we care about
+    AllKinds: TypeAlias = Union[
+        ZeroBasedOpenEndedProtocol, OneBasedProtocol, ZeroBasedUnstranded, ZeroBasedStranded
+    ]
+    features: List[AllKinds] = [
+        zero_based_protocol,
+        one_based_protocol,
+        zero_based_unstranded,
+        zero_based_stranded,
+    ]
+    detector: OverlapDetector[AllKinds] = OverlapDetector(features)
+
+    assert OverlapDetector._negative(zero_based_protocol) is False
+    assert OverlapDetector._negative(one_based_protocol) is False
+    assert OverlapDetector._negative(zero_based_unstranded) is False
+    assert OverlapDetector._negative(zero_based_stranded) is True
+
+    # Query the overlap detector with yet another type
+    assert detector.get_overlaps(Interval("chr1", 0, 1)) == []
+    assert detector.get_overlaps(Interval("chr1", 0, 9)) == [zero_based_protocol]
+    assert detector.get_overlaps(Interval("chr1", 11, 12)) == [
+        zero_based_protocol,
+        one_based_protocol,
+    ]
+    assert detector.get_overlaps(Interval("chr1", 21, 27)) == [
+        zero_based_protocol,
+        one_based_protocol,
+        zero_based_unstranded,
+    ]
+    assert detector.get_overlaps(Interval("chr1", 32, 35)) == [
+        zero_based_protocol,
+        one_based_protocol,
+        zero_based_unstranded,
+        zero_based_stranded,
+    ]
+    assert detector.get_overlaps(Interval("chr1", 54, 55)) == [
+        one_based_protocol,
+        zero_based_unstranded,
+        zero_based_stranded,
+    ]
+    assert detector.get_overlaps(Interval("chr1", 61, 62)) == [
+        zero_based_unstranded,
+        zero_based_stranded,
+    ]
+    assert detector.get_overlaps(Interval("chr1", 78, 79)) == [zero_based_stranded]
+    assert detector.get_overlaps(Interval("chr1", 80, 81)) == []
```

Comment:
somewhere here I'd like to test (via mypy) that when we type hint the return types, mypy is ok with it.  Something like:

```python
    retvals: list[AllKinds] = detector.get_overlaps(Interval("chr1", 78, 79))

    detector: OverlapDetector[OneBasedProtocol]  = OverlapDetector(one_based_protocol)
    retvals: list[ OneBasedProtocol]  = detector.get_overlaps(one_based_protocol)
    assert retvals == [one_based_protocol]
```